shared-libs/cht-datasource/src/index.ts

```diff
@@ -54,20 +64,73 @@ export const getDatasource = (ctx: DataContext) => {
     v1: {
       hasPermissions,
       hasAnyPermission,
+      contact: {
+        /**
+         * Returns a contact by their UUID.
+         * @param uuid the UUID of the contact to retrieve
+         * @returns the contact or `null` if no contact is found for the UUID
+         * @throws InvalidArgumentError if no UUID is provided
+         */
+        getByUuid: (uuid: string) => ctx.bind(Contact.v1.get)(Qualifier.byUuid(uuid)),
+
+        /**
+         * Returns a contact by their UUID along with the contact's parent lineage.
+         * @param uuid the UUID of the contact to retrieve
+         * @returns the contact or `null` if no contact is found the UUID
+         * @throws InvalidArgumentError if no UUID is provided
+         */
+        getByUuidWithLineage: (uuid: string) => ctx.bind(Contact.v1.getWithLineage)(Qualifier.byUuid(uuid)),
+
+        /**
+         * Returns an array of contact identifiers for the provided page specifications.
+         * @param freetext the search keyword(s)
+         * @param type the type of contact to search for
+         * @param cursor the token identifying which page to retrieve. A `null` value indicates the first page should be
+         * returned. Subsequent pages can be retrieved by providing the cursor returned with the previous page.
+         * @param limit the maximum number of identifiers to return. Default is 10000.
+         * @returns a page of contact identifiers for the provided specifications
+         * @throws InvalidArgumentError if either `freetext` or `type` is not provided
+         * @throws InvalidArgumentError if the `freetext` is empty or if the `type is invalid for a contact
+         * @throws InvalidArgumentError if the provided limit is `<= 0`
+         * @throws InvalidArgumentError if the provided cursor is not a valid page token or `null`
+         */
+        getIdsPage: (
+          freetext: Nullable<string> = null,
+          type: Nullable<string> = null,
```

Comment:
I agree with combining the qualifiers part but not the naming part. Suppose we now name it `getIdsPageByTypeFreetext` and add a param as you said. Would we rename it?

Reply:
My thought would be that if we have separate functions for each type of search, then when we go to support another type of search (e.g. get-contact-ids-by-parent or something like that) we would just add a new function instead of adding a new parameter to the existing functions.